static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx

```diff
@@ -0,0 +1,240 @@
+import {useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+import {PlatformIcon} from 'platformicons';
+
+import {Button} from 'sentry/components/button';
+import EmptyStateWarning from 'sentry/components/emptyStateWarning';
+import Link from 'sentry/components/links/link';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {Panel} from 'sentry/components/panels';
+import PerformanceDuration from 'sentry/components/performanceDuration';
+import {Flex} from 'sentry/components/profiling/flex';
+import {
+  FunctionsMiniGrid,
+  FunctionsMiniGridEmptyState,
+  FunctionsMiniGridLoading,
+} from 'sentry/components/profiling/functionsMiniGrid';
+import {TextTruncateOverflow} from 'sentry/components/profiling/textTruncateOverflow';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization, Project} from 'sentry/types';
+import {
+  EventsResultsDataRow,
+  useProfileEvents,
+} from 'sentry/utils/profiling/hooks/useProfileEvents';
+import {useProfilingTransactionQuickSummary} from 'sentry/utils/profiling/hooks/useProfilingTransactionQuickSummary';
+import {generateProfileSummaryRoute} from 'sentry/utils/profiling/routes';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+
+const fields = ['transaction', 'project.id', 'last_seen()', 'p95()', 'count()'] as const;
+type SlowestTransactionsFields = typeof fields[number];
+
+export function ProfilingSlowestTransactionsPanel() {
+  const profilingTransactionsQuery = useProfileEvents({
+    fields,
+    sort: {
+      key: 'p95()',
+      order: 'desc',
+    },
+    limit: 3,
+    query: 'count():>3',
+    referrer: 'api.profiling.landing-slowest-transaction-panel',
+  });
+
+  const [openPanel, setOpenPanel] = useState<null | string>(null);
+
+  const profilingTransactions = useMemo(
+    () => profilingTransactionsQuery.data?.[0].data ?? [],
+    [profilingTransactionsQuery.data]
+  );
+
+  const transactionNames = useMemo(
+    () => profilingTransactions.map(txn => txn.transaction),
+    [profilingTransactions]
+  );
+
+  if (transactionNames.length > 0 && !transactionNames.includes(openPanel)) {
+    const firstTransaction = transactionNames[0];
+    setOpenPanel(firstTransaction as string);
```

Comment:
feel free to just setOpenPanel(transactionNames[0]!) in these cases, it's readable enough imo

Reply:
I think you are also setting this to initialize the first state right?

Reply:
yeah i am initializing state here... seeing how far i can get with this inline `setState` pattern to see where the footgun is